web/source/osk/activeLayout.ts

```diff
@@ -55,24 +88,39 @@ namespace com.keyman.osk {
         keys[j]['widthpc']=keyPercent;
         padPercent=parseInt(keys[j]['pad'],10)/totalWidth;
         keys[j]['padpc']=padPercent;
+
+        // compute center's default x-coord
+        (<ActiveKey> keys[j]).proportionalX = (totalPercent + padPercent + (keyPercent/2));
+        (<ActiveKey> keys[j]).proportionalWidth = keyPercent;
+
         totalPercent += padPercent+keyPercent;
       }
 
       // Allow for right OSK margin (15 layout units)
-      totalPercent += 15/totalWidth;
+      let rightMargin = 15/totalWidth;
+      totalPercent += rightMargin;
 
       // If a single key, and padding is negative, add padding to right align the key
       if(keys.length == 1 && parseInt(keys[0]['pad'],10) < 0) {
         keyPercent=parseInt(keys[0]['width'],10)/totalWidth;
         keys[0]['widthpc']=keyPercent;
         totalPercent += keyPercent;
         keys[0]['padpc']=1-totalPercent;
+
+        // compute center's default x-coord
+        (<ActiveKey> keys[0]).proportionalX = ((totalPercent - rightMargin) -  keyPercent/2);
+        (<ActiveKey> keys[0]).proportionalWidth = keyPercent;
```

Comment:
A helper function like `computeCenterDefaultXSingleKey()` could be useful here to avoid code duplication.